utils/types.go

```diff
@@ -80,6 +80,18 @@ func (e ErrArray) Error() string {
 	return res
 }
 
+func (e ErrArray) ErrorList() []string {
+	errList := []string{}
+	if len(e.Errors) == 0 {
```

Comment:
Is this check needed? Since you're looping over `e.Errors`, if it's empty the logic in the loop will not be executed.

```
func (e ErrArray) ErrorList() []string {
	errList := []string{}
	for _, err := range e.Errors {
		errList = append(errList, fmt.Sprintf("%v", err)) 
	}
	return errList
}
```